platform/darwin/src/MGLRendererConfiguration.mm

```diff
@@ -1,19 +1,62 @@
 #import "MGLRendererConfiguration.h"
 #import "MGLOfflineStorage_Private.h"
+#import "MGLFoundation_Private.h"
 
 #if TARGET_OS_IPHONE
 #import <UIKit/UIKit.h>
 #else
 #import <AppKit/AppKit.h>
 #endif
 
+static NSString * const MGLCollisionBehaviorPre4_0Key = @"MGLCollisionBehaviorPre4_0";
+
+@interface MGLRendererConfiguration ()
+@property (nonatomic, readwrite) BOOL perSourceCollisions;
+@end
+
 
 @implementation MGLRendererConfiguration
 
 + (instancetype)currentConfiguration {
     return [[self alloc] init];
```

Comment:
With the addition of the initializers I think we need to make this class a proper singleton. Or is there a reason to have more than one instance for this class?

Reply:
I'm not aware of the history behind `currentConfiguration`, but I think it's like this so that it captures the current state, which could change. I agree it's a little confusing - hence my comment above:
> (An aside: since currentConfiguration doesn't do anything I'm tempted to remove it too).